### PR TITLE
SignIn form disabled when reverse proxy auth is enabled

### PR DESCRIPTION
**Pull request by pboguslawski**
SignIn form should not be enabled when users are authenticated
with reverse proxy.

Author-Change-Id: IB#1115398

### CODE REVIEW

File: routers/web/web.go [hunk 1]
```diff
@@ -225,6 +225,14 @@ func RegisterRoutes(m *web.Route) {
 		}
 	}
 
+	// reverseProxyAuthDisabled rquires reverse proxy authentication to be disabled by admin.
```
Comment (silverwind) [inline routers/web/web.go]:
```suggestion
	// reverseProxyAuthDisabled requires reverse proxy authentication to be disabled by admin.
```

File: modules/templates/helper.go [hunk 2]
```diff
@@ -240,6 +240,9 @@ func NewFuncMap() []template.FuncMap {
 		"DisableImportLocal": func() bool {
 			return !setting.ImportLocalPaths
 		},
+		"DisableReverseProxyAuth": func() bool {
```
Comment (lunny) [inline modules/templates/helper.go:243]: Maybe it's better to use `EnableReverseProxyAuth`?

**Comment by pboguslawski** [comment]
`DisableGitHooks`
`DisableWebhooks`
`DisableImportLocal`
...so why not `DisableReverseProxyAuth`?

**Comment by wxiaoguang** [comment]
As a personal/private usage, these changes are good.

As a general feature for all users, I am not sure whether these changes are complete.

For example, there are other handlers besides "login": "openid", "sign_up", "oauth2", etc.

To make it a general feature for all users (avoid any misuse or risk), I think there should be a whole design/plan for the Account System first.

Just my personal opinion, correct me if I was wrong.

**Comment by pboguslawski** [comment]
> As a personal/private usage, these changes are good.
> As a general feature for all users, I am not sure whether these changes are complete.

Disabling unused auth stuff is good for professional usage also IHMO.

> To make it a general feature for all users (avoid any misuse or risk), I think there should be a whole design/plan for the Account System first.

Gitea should implement pluging-based auth system and plugin-based authz system IHMO and not focus on specific auth soluition (like tokens/passwords, etc.). Separate auth/authz plugin config for front and API probably if separate set of APIs is necessary. Not easy to implement probably so consider spliiting this idea to separate thread to discuss and implement. This mod would be obsolete if signing screen was part of "password auth plugin".

**Comment by wxiaoguang** [comment]
It should have been fixed by  Allow to disable the password-based login (sign-in) form #32687

**Pull request closed without merging**